### PR TITLE
Updating concepts to CIEL 20181210 and preparing for next release

### PR DESCRIPTION
**Pull request by reubenvarghese1**
Updating concepts to CIEL 20181210 and preparing for next release

### CODE REVIEW

**Comment by reubenvarghese1** [comment]
@dkayiwa It would be great if you could review this PR so that it may be merged(Don't want to take chances by pushing without a review) :)

File: api/src/main/resources/Reference_Application_Concepts-23.xml [hunk 1]
```diff
@@ -231,123 +239,123 @@
...
+  <concept_name concept_id="1067" name="জানিনা" locale="bn" creator="1" date_created="2015-04-24 17:19:36.0" concept_name_id="127334" voided="false" voided_by="1" uuid="127334BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_name_type="FULLY_SPECIFIED" locale_preferred="true"/>
   <concept_name concept_id="1067" name="Enkoni" locale="ht" creator="1" date_created="2016-02-20 05:13:35.0" concept_name_id="135594" voided="false" voided_by="1" uuid="135594BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_name_type="FULLY_SPECIFIED" locale_preferred="true"/>
-  <concept_name concept_id="1067" name="&#1605;&#1593;&#1604;&#1608;&#1605; &#1606;&#1729;&#1740;&#1722;" locale="ur" creator="1" date_created="2016-08-16 07:46:02.0" concept_name_id="139801" voided="false" voided_by="1" uuid="139801BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_name_type="FULLY_SPECIFIED" locale_preferred="true"/>
+  <concept_name concept_id="1067" name="معلوم نہیں" locale="ur" creator="1" date_created="2016-08-16 07:46:02.0" concept_name_id="139801" voided="false" voided_by="1" uuid="139801BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_name_type="FULLY_SPECIFIED" locale_preferred="true"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Concepts-23.xml]: Did you intentionally make changes to this file?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: I just copied the output of the export tool. I guess chrome maybe converted it automatically?

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: Do we need to change anything in this file?

File: api/src/main/resources/Reference_Application_Concepts-23.xml [hunk 2]
```diff
@@ -115,34 +115,42 @@
   <concept concept_id="162821" retired="false" datatype_id="4" class_id="11" is_set="false" creator="1" date_created="2015-01-06 22:32:39.0" version="" changed_by="1" date_changed="2015-01-06 22:32:47.0" retired_by="1" uuid="162821AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
   <concept concept_id="162822" retired="false" datatype_id="4" class_id="11" is_set="false" creator="1" date_created="2015-01-06 22:33:22.0" version="" changed_by="1" date_changed="2015-01-06 22:33:29.0" retired_by="1" uuid="162822AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
   <concept concept_id="163139" retired="false" datatype_id="4" class_id="11" is_set="false" creator="1" date_created="2015-12-13 15:00:18.0" version="" changed_by="1" date_changed="2015-12-13 15:02:03.0" retired_by="1" uuid="163139AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
-  <concept concept_id="165095" retired="false" datatype_id="3" class_id="7" is_set="false" creator="1" date_created="2018-04-18 14:34:20.0" version="" changed_by="1" date_changed="2018-04-18 14:34:30.0" retired_by="1" uuid="165095AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept_numeric concept_id="1343" units="cm" precise="true"/>
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Concepts-23.xml]: Maybe this needs to be added @dkayiwa ?

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: What happens if it is not added?

File: api/src/main/resources/Reference_Application_Concepts-23.xml [hunk 3]
```diff
@@ -36,15 +36,15 @@
...
-  <concept concept_id="77419" retired="false" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2006-12-17 00:00:00.0" version="" changed_by="1" date_changed="2006-12-17 00:00:00.0" retired_by="1" uuid="77419AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
-  <concept concept_id="80106" retired="false" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2006-12-17 00:00:00.0" version="" changed_by="1" date_changed="2015-11-05 01:56:42.0" retired_by="1" uuid="80106AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
-  <concept concept_id="84462" retired="false" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2006-12-17 00:00:00.0" version="" changed_by="1" date_changed="2006-12-17 00:00:00.0" retired_by="1" uuid="84462AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept concept_id="71617" retired="false" form_text="" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2006-12-17 00:00:00.0" version="" changed_by="1" date_changed="2016-07-29 22:19:54.0" retired_by="1" uuid="71617AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Concepts-23.xml]: Here, this "form text" is something that @rkorytkowski had deleted in a previous commit : https://github.com/openmrs/openmrs-module-referencemetadata/commit/367afa385676a0cadfe5d5f3cbc3ec0a60792d8d . These are being generated automatically yet again by the Data Export tool

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: I understand. Have you looked at the actual changes that you have made to this file?

File: api/src/main/resources/Reference_Application_Concepts-23.xml (hunk 2 above)
Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: To be honest, I don't really have much of a clue about the implications of having added(or not having added) this line. The reason being that I just copied the output from the tool as mentioned in the Wiki

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: Am not specifically referring to this line. I mean all changes in this file. Did you take a look at each of them?

**Comment by dkayiwa** [comment]
@reubenvarghese1 did you see the travis failure?

File: api/src/main/resources/Reference_Application_Concepts-23.xml (hunk 3 above)
Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: @dkayiwa  After giving it a proper read, I can conclude that the only "real" changes made to this file are the removal of the concept with ID=165095

**Comment by dkayiwa** [comment]
@reubenvarghese1 did you see the travis build failure?

**Comment by reubenvarghese1** [comment]
@dkayiwa  Yes I did, I am unable to figure out the cause for it though. I tried removing the form_name column but that doesn't seem to do the trick

**Comment by dkayiwa** [comment]
Does it also fail locally on your computer?

**Comment by reubenvarghese1** [comment]
@dkayiwa Yes, after I put in my changes it begins to fail. I do not see anything out of the ordinary that didn't exist before though

**Comment by reubenvarghese1** [comment]
@dkayiwa  I think I know why the error occurs. Actually, in this commit, one of the concepts with the id 165095 has been removed. So maybe the test should now assert that it is 439 and not 440?

**Comment by reubenvarghese1** [comment]
@dkayiwa I think my suspicion is right. On changing it to 439, the build succeeds. Do you suggest I modify the test as well? Or , I can also push the changes for you to review :)

**Comment by dkayiwa** [comment]
A pull request represents code that is not yet merged. You therefore do not need to first ask permission in order to try out anything that crosses your mind. It can always be reviewed and reverted, if necessary.

**Comment by reubenvarghese1** [comment]
@dkayiwa The failure that I get is because of some formatting issue in the xml files(As in, certain columns or something are messing with the structure of the xml file . For eg: the ' retired_by="1" ' column(in the Order_Entry_and_Allergies_Concepts-18.xml). I managed to resolve the errors in the other two files. The build failure is now caused by the Order_Entry_and_Allergies_Concepts-18.xml file. I can verify this because, if I revert to the older file, the build passes. 

Any idea on what may be the issue?

**Comment by dkayiwa** [comment]
Are all your changes committed?
Did you you intentionally make any changes to Reference_Application_Diagnoses-10.xml and Order_Entry_and_Allergies_Concepts-17.xml?

**Comment by reubenvarghese1** [comment]
@dkayiwa Yes, I intentionally renamed them and updated their contents as mentioned in the reference application release process wiki

**Comment by dkayiwa** [comment]
They look to be very very many changes in those two files. Did you intentionally change all those?

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml [hunk 4]
```diff
@@ -1,240 +1,240 @@
...
+  <concept_class concept_class_id="11" name="Misc" description="Terms which don&apos;t fit other categories" creator="1" date_created="2004-03-02 00:00:00.0" retired="false" retired_by="1" uuid="8d492774-c2cc-11de-8d13-0010c6dffd0f"/>
+  <concept_class concept_class_id="12" name="Symptom" description="Patient-reported observation" creator="1" date_created="2004-10-04 00:00:00.0" retired="false" retired_by="1" uuid="8d492954-c2cc-11de-8d13-0010c6dffd0f"/>
+  <concept_class concept_class_id="13" name="Symptom/Finding" description="Observation that can be reported from patient or found on exam" creator="1" date_created="2004-10-04 00:00:00.0" retired="false" retired_by="1" uuid="8d492b2a-c2cc-11de-8d13-0010c6dffd0f"/>
+  <concept concept_id="48" retired="false" datatype_id="2" class_id="4" is_set="false" creator="1" date_created="2004-01-01 00:00:00.0" version="" changed_by="1" date_changed="2012-10-09 21:18:04.0" retired_by="1" uuid="48AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Diagnoses-11.xml]: @dkayiwa  These changes are intentional. I mean the contents of this file was generated using the mdsbuilder. If you notice, most of these changes are involving the addition of a new attribute ("retired_by="1""). Don't know whether this is needed. I have tried removing these small changes and running the build locally. It still fails. Apart from this, it is worth noting that there are a few genuine additions to this file as well.

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml [hunk 5]
```diff
@@ -3231,6 +3231,7 @@
   <concept_reference_map concept_map_id="143778" creator="1" date_created="2011-06-08 00:29:05.0" concept_id="122604" uuid="143778ABBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_reference_term_id="134058" concept_map_type_id="2" changed_by="1"/>
   <concept_reference_map concept_map_id="186149" creator="1" date_created="2013-07-03 18:53:08.0" concept_id="122604" uuid="186149ABBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_reference_term_id="186149" concept_map_type_id="1" changed_by="1"/>
   <concept_reference_map concept_map_id="268810" creator="1" date_created="2013-10-01 13:54:05.0" concept_id="122604" uuid="268810ABBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_reference_term_id="268810" concept_map_type_id="1" changed_by="1"/>
+  <concept_reference_map concept_map_id="283353" creator="1" date_created="2018-05-01 16:26:59.0" concept_id="122604" uuid="283353ABBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_reference_term_id="159009" concept_map_type_id="2" changed_by="1"/>
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Example of the addition of a new concept

**Comment by reubenvarghese1** [comment]
Hi @dkayiwa, do you have any suggestions on how to proceed?

**Comment by dkayiwa** [comment]
@reubenvarghese1 did you look at the error message of the failing test?

**Comment by reubenvarghese1** [comment]
@dkayiwa Yes, I did.  org.dbunit.dataset.NoSuchColumnException. There could be many reasons for this error. However, since my changes are trivial, I am not able to understand the reason for this error's occurrence. I believe that it occurs due to an inconsistency in the format of the xml file but I do not know where the inconsistency lay.

**Comment by dkayiwa** [comment]
@reubenvarghese1 did you find out the actual column and table?

**Comment by reubenvarghese1** [comment]
@dkayiwa, yes. It says form_text column is the issue. However there is no such attribute even mentioned in the xml files. Like i mentioned, the issue is in the Order_Entry_and_Allergies_Concepts file(some formatting issue since without changing this file, the build passes. The same error existed with the other 2 modified xml files too but i managed to fix their formatting)

**Comment by dkayiwa** [comment]
@reubenvarghese1 did you try googling with that error message text? java.lang.RuntimeException: org.dbunit.dataset.NoSuchColumnException: CONCEPT.FORM_TEXT

**Comment by reubenvarghese1** [comment]
@dkayiwa Cloned it again, made all the changes again. Now it works! Kindly check it out now :)

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml [hunk 6]
```diff
@@ -1,240 +1,240 @@
 <?xml version='1.0' encoding='UTF-8'?>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Where do we use this file?

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 7]
```diff
@@ -1,216 +1,217 @@
 <?xml version='1.0' encoding='UTF-8'?>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: Where do we use this file?

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml (hunk 6 above)
Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: @dkayiwa I am not sure tbh. It is mentioned as a step in the [release process wiki](https://wiki.openmrs.org/display/docs/OpenMRS+Reference+Application+Release+Process) .
The step states the following:
"Go to Administration -> Metadata Sharing -> Export Metadata and choose a package (repeat b. and c. for the following packages: Reference Application Concepts, Reference Application Diagnoses, Reference Application Order Entry and Allergies Concepts)"

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: That is why I made these changes

File: api/src/main/resources/Reference_Application_Concepts-23.xml [hunk 8]
```diff
@@ -835,11 +843,10 @@
...
-  <concept_name concept_id="165095" name="General patient note" locale="en" creator="1" date_created="2018-04-18 14:34:20.0" concept_name_id="141526" voided="false" voided_by="1" uuid="141526BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" concept_name_type="FULLY_SPECIFIED" locale_preferred="true"/>
-  <concept_map_type concept_map_type_id="1" name="SAME-AS" creator="1" date_created="2018-05-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false" retired_by="1" uuid="35543629-7d8c-11e1-909d-c80aa9edcf4e"/>
-  <concept_map_type concept_map_type_id="2" name="NARROWER-THAN" creator="1" date_created="2018-05-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false" retired_by="1" uuid="43ac5109-7d8c-11e1-909d-c80aa9edcf4e"/>
-  <concept_map_type concept_map_type_id="3" name="BROADER-THAN" creator="1" date_created="2018-05-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false" retired_by="1" uuid="4b9d9421-7d8c-11e1-909d-c80aa9edcf4e"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Concepts-23.xml]: Is this just a formatting change? I do not see any difference between the red and green lines.

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: Can we fix the dates to be the same?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: @dkayiwa Reverted those lines back to the previous one

**Comment by reubenvarghese1** [comment]
Hi @dkayiwa , are the latest changes looking good to you?

**Comment by dkayiwa** [comment]
Do you know why there are so so so and so many changes for the Reference_Application_Diagnoses-11.xml and Order_Entry_and_Allergies_Concepts-18.xml files?

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml [hunk 9]
```diff
@@ -1,240 +1,240 @@
...
+  <concept concept_id="110834" retired="false" datatype_id="4" class_id="4" is_set="false" creator="1" date_created="2007-11-03 00:00:00.0" version="" changed_by="1" date_changed="2015-11-03 02:56:58.0" retired_by="1" uuid="110834AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept concept_id="111633" retired="false" datatype_id="4" class_id="4" is_set="false" creator="1" date_created="2007-11-03 00:00:00.0" version="" changed_by="1" date_changed="2015-11-03 02:53:04.0" retired_by="1" uuid="111633AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept concept_id="111636" retired="false" datatype_id="2" class_id="4" is_set="false" creator="1" date_created="2007-11-03 00:00:00.0" version="" changed_by="1" date_changed="2016-11-27 22:21:19.0" retired_by="1" uuid="111636AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept concept_id="112198" retired="false" datatype_id="4" class_id="4" is_set="false" creator="1" date_created="2007-11-03 00:00:00.0" v
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Diagnoses-11.xml]: @dkayiwa  A whole bunch of these changes is due to this new column called retired_by. Should we have it?

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Why would be need a retired_by when retired="false"?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Exactly

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Exactly

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: I guess we'll remove it

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: So can you?

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml [hunk 10]
```diff
@@ -2473,7 +2473,7 @@
   <concept_reference_term concept_reference_term_id="275612" concept_source_id="13" code="62" creator="1" date_created="2014-05-12 00:00:00.0" changed_by="1" retired="false"  uuid="dc5868f6-c78a-3818-8ebd-224da4ba342d"/>
   <concept_reference_term concept_reference_term_id="275613" concept_source_id="13" code="66" creator="1" date_created="2014-05-12 00:00:00.0" changed_by="1" retired="false"  uuid="d0196c86-7bd9-3bdf-931c-2d3d2653bc6b"/>
   <concept_reference_term concept_reference_term_id="275615" concept_source_id="13" code="82" creator="1" date_created="2014-05-12 00:00:00.0" changed_by="1" retired="false"  uuid="c2086097-9ba0-3061-a249-c784d7e0b14b"/>
-  <concept_reference_term concept_reference_term_id="275617" concept_source_id="13" code="122" creator="1" date_created="2014-05-12 00:00:00.0" changed_by="1" retired="false" uuid="fe59f3fc-b1f9-3b79-974b-81dd7698d9cf"/>
+  <concept_reference_term concept_reference_term_id="275617" concept_source_id="13" code="122" creator="1" date_created="2014-05-12 00:00:00.0" changed_by="1" retired="false"  uuid="fe59f3fc-b1f9-3b79-974b-81dd7698d9cf"/>
```
Comment (reubenvarghese1) [inline api/src/main/resources/Reference_Application_Diagnoses-11.xml]: @dkayiwa , I have intentionally let these spaces be as they are making the format uniform. Most of the other lines of the similar type have two spaces. These lines had just a single space, so I have added the additional space to make it uniform. Let me know if you feel this shouldn't be

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml (hunk 9 above)
Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: @dkayiwa Yes, was working on it. Just committed the changes. Do check out :)

File: api/src/main/resources/Reference_Application_Diagnoses-11.xml (hunk 10 above)
Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Diagnoses-11.xml]: Can we use a separate pull request for formatting changes? It becomes hard to review and quickly merge a pull request which has a mixture of actual and formatting changes.

**Comment by dkayiwa** [comment]
I still see lots of changes in the two files Reference_Application_Diagnoses-11.xml and Order_Entry_and_Allergies_Concepts-18.xml files? Are they all necessary changes?

**Comment by reubenvarghese1** [comment]
@dkayiwa  Removed as many frivolous changes as I could. Order_Entry_and_Allergies_Concepts-18.xml  has some significant changes

**Comment by reubenvarghese1** [comment]
@dkayiwa , this is a reminder in case you missed my previous comment

**Comment by dkayiwa** [comment]
Can we also reduce the number of changes in the file Order_Entry_and_Allergies_Concepts-17.xml ?

**Comment by reubenvarghese1** [comment]
@dkayiwa , I don't see anything in that file that can be reduced further. I have  already omitted a few things which in my opinion weren't necessary

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 11]
```diff
@@ -18,7 +18,7 @@
...
   <concept concept_id="121" retired="false" datatype_id="4" class_id="5" is_set="false" creator="1" date_created="2004-01-01 00:00:00.0" version="" changed_by="1" date_changed="2015-11-05 01:37:39.0"  uuid="121AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
   <concept concept_id="512" retired="false" datatype_id="2" class_id="13" is_set="false" creator="1" date_created="2004-01-01 00:00:00.0" version="596285" changed_by="1" date_changed="2016-04-27 14:18:52.0"  uuid="512AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
-  <concept concept_id="924" retired="false" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2004-07-23 00:00:00.0" version="" changed_by="1" date_changed="2010-10-01 14:33:22.0"  uuid="924AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept concept_id="924" retired="false" datatype_id="4" class_id="3" is_set="false" creator="1" date_created="2004-07-23 00:00:00.0" version="" changed_by="1" date_changed="2017-12-14 12:34:09.0"  uuid="924AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: What was changed on the above line to necessitate a date_changed change?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: Nothing. Why does the date change happen without any perceptible changes?

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: So, can we avoid such changes of only the date_changed?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: Yes sure, I am already on it. Will push a commit in a few mins :)

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 12]
```diff
@@ -2334,46 +2357,48 @@
...
-  <concept_set concept_set_id="1648" concept_id="162364" concept_set="162384" sort_weight="17.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1648AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
-  <concept_set concept_set_id="1649" concept_id="162365" concept_set="162384" sort_weight="18.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1649AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
-  <concept_set concept_set_id="1650" concept_id="162366" concept_set="162384" sort_weight="19.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1650AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
-  <concept_set concept_set_id="1651" concept_id="161553" concept_set="162384" sort_weight="20.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1651AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
-  <concept_set concept_set_id="1652" concept_id="162263" concept_set="162384" sort_weight="21.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1652AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
-  <concept_set concept_set_id="1653" concept_id="162367" concept_set="162384" sort_weight="22.0" creator="1" date_created="2014-05-15 20:36:58.0" uuid="1653AEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: Why did the sort_weights change?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: I'm not quite sure. This is what was generated by the data exchange module. Infact, all the changes so far were what was got as output

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 13]
```diff
@@ -207,10 +207,11 @@
...
   <concept_description concept_description_id="120" concept_id="121" description="Muscular pain or tenderness, especially when diffuse and nonspecific." locale="en" creator="1" date_created="2004-01-01 00:00:00.0" changed_by="1" date_changed="2015-11-05 01:37:39.0" uuid="120FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"/>
   <concept_description concept_description_id="512" concept_id="512" description="DEF: Lesion of the skin. DESC: The history of the particular skin lesions is also important, including time and site of initial appearance, spread, change in appearance, and triggering factors. Visual inspection is the central evaluation tool; many skin diseases are diagnosed by the characteristic appearance or morphology of the lesions. REF: Merck Manual" locale="en" creator="1" date_created="2004-01-01 00:00:00.0" changed_by="1" date_changed="2016-04-27 14:18:52.0" uuid="512FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"/>
-  <concept_description concept_description_id="927" concept_id="924" description="Antimalarial prophylactic and therapy.  Also being used for PCP treatment." locale="en" creator="1" date_created="2004-07-23 00:00:00.0" changed_by="1" uuid="927FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: What actually changed here to necessitate a date_changed change?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: @dkayiwa I guess maybe the fact that changed_by is 1 and that this line didn't have a date_changed column before makes the addition of the column justified?

Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: changed_by without any actual change?

**Comment by reubenvarghese1** [comment]
@dkayiwa Removed that change

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 14]
```diff
@@ -1186,28 +1188,32 @@
...
   <concept_name concept_id="162824" name="Intraosseous" locale="en" creator="1" date_created="2015-01-12 18:19:41.0" concept_name_id="126874" voided="false" voided_by="1" uuid="126874BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" locale_preferred="true"/>
-  <concept_map_type concept_map_type_id="1" name="SAME-AS" creator="1" date_created="2017-11-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false"  uuid="35543629-7d8c-11e1-909d-c80aa9edcf4e"/>
-  <concept_map_type concept_map_type_id="2" name="NARROWER-THAN" creator="1" date_created="2017-11-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false"  uuid="43ac5109-7d8c-11e1-909d-c80aa9edcf4e"/>
-  <concept_map_type concept_map_type_id="3" name="BROADER-THAN" creator="1" date_created="2017-11-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false"  uuid="4b9d9421-7d8c-11e1-909d-c80aa9edcf4e"/>
-  <concept_map_type concept_map_type_id="4" name="Associated finding" creator="1" date_created="2017-11-03 00:00:00.0" changed_by="1" is_hidden="false" retired="false"  uuid="55e02065-7d8c-11e1-909d-c80aa9edcf4e"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: What has actually changed in these lines?

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml [hunk 15]
```diff
@@ -1186,28 +1188,32 @@
...
-  <concept_reference_source concept_source_id="18" name="HL7 2.x Route of Administration" description="Members of value set from HL7 for routes of administration" hl7_code="HL70162" creator="1" date_created="2012-10-19 13:43:57.0" retired="false"  uuid="98e77229-19e7-45e0-88dd-e30214dd2fad"/>
-  <concept_reference_source concept_source_id="19" name="3BT" description="Belgian Bilingual Bi-encoded Thesaurus (3BT)" creator="1" date_created="2013-02-16 22:25:25.0" retired="false"  uuid="ef7f179f-d97d-4a8c-994c-67c544428684"/>
-  <concept_reference_source concept_source_id="20" name="ICPC2" description="International Classification of Primary Care version 2 from WONCA." creator="1" date_created="2013-02-17 15:25:41.0" retired="false"  uuid="32fb691d-928b-48a9-b3ee-ca64f21f20bf"/>
-  <concept_reference_source concept_source_id="21" name="CIEL" description="Columbia International eHealth Laboratory concept ID" creator="1" date_created="2013-07-03 15:55:03.0" retired="false"  uuid="249b13c8-72fa-4b96-8d3d-b200efed985e"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: What has actually changed in these lines?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: @dkayiwa , is it really necessary to manually revert all the changes? I have already changed so many dates. All of them have been modified by the data exchange module. Doesn't the amount of manual effort defeat the purpose of the module?

**Comment by reubenvarghese1** [comment]
@dkayiwa , This is a gentle reminder in case you missed my previous comment

File: api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml (hunk 15 above)
Comment (dkayiwa) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: Does this mean that the dates had changed from the last time an export was done?

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Order_Entry_and_Allergies_Concepts-18.xml]: @dkayiwa , here I guess the major change is the modification of the UUID and not the dates.

**Comment by reubenvarghese1** [comment]
@dkayiwa , This is a gentle reminder in case you missed my previous comment

File: api/src/main/resources/Reference_Application_Concepts-23.xml [hunk 16]
```diff
@@ -115,7 +115,15 @@
...
   <concept concept_id="163139" retired="false" datatype_id="4" class_id="11" is_set="false" creator="1" date_created="2015-12-13 15:00:18.0" version="" changed_by="1" date_changed="2015-12-13 15:02:03.0" retired_by="1" uuid="163139AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
-  <concept concept_id="165095" retired="false" datatype_id="3" class_id="7" is_set="false" creator="1" date_created="2018-04-18 14:34:20.0" version="" changed_by="1" date_changed="2018-04-18 14:34:30.0" retired_by="1" uuid="165095AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"/>
+  <concept_numeric concept_id="1343" units="cm" precise="true"/>
+  <concept_numeric concept_id="5085" hi_absolute="250.0" low_absolute="0.0" units="mmHg" precise="false"/>
+  <concept_numeric concept_id="5086" hi_absolute="150.0" low_absolute="0.0" units="mmHg" precise="false"/>
+  <concept_numeric concept_id="5087" hi_absolute="230.0" low_absolute="0.0" units="beats/min" precise="false"/>
+  <concept_numeric concept_id="5088" hi_absolute="43.0" low_absolute="25.0" units="DEG C" precise="true"/>
```
Comment (dkayiwa) [inline api/src/main/resources/Reference_Application_Concepts-23.xml]: According to this commit https://github.com/openmrs/openmrs-module-referencemetadata/commit/9f0a937e2d7975b5edd0a842b4f085db0d20f6d1 the concept numeric rows should be entered in different file names.

Comment (reubenvarghese1) [reply api/src/main/resources/Reference_Application_Concepts-23.xml]: From the output that I had got from the data exchange module. Everything in the 3 XML files have been generated by the module ( including the dates and anything else)

**Comment by reubenvarghese1** [comment]
@dkayiwa , This is a gentle reminder in case you missed my previous comment

**Comment by reubenvarghese1** [comment]
Yes exactly @dkayiwa

**Pull request closed without merging**